File: 735-asteroid-collision/asteroid-collision.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        arr = []
        i = 0
        while(i < len(asteroids)):
            curr = asteroids[i]
            inside = 0
            if curr < 0:
                while(arr and arr[-1] > 0 and -curr > arr[-1]):
                    arr.pop(-1)
                if arr and -curr == arr[-1]:
                    arr.pop(-1)
                elif not arr or (arr and arr[-1] < 0):
                    arr.append(curr)
            else:
                arr.append(curr)
            i+=1
        return arr

File: 735-asteroid-collision/asteroid-collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> stack = new Stack<>();
        for(int i = 0;i<asteroids.length;i++){
            if(stack.isEmpty()){
                stack.push(asteroids[i]);
            }
            else{
                if(asteroids[i] < 0){
                    while(!stack.isEmpty() && stack.peek() > 0 && stack.peek() < -1*asteroids[i]){
                        stack.pop();
                    }
                    if(!stack.isEmpty() && stack.peek() == -1 * asteroids[i]){
                        stack.pop();
                    }
                    else if(stack.isEmpty() || stack.peek() < 0){
                        stack.push(asteroids[i]);
                    }
                }
                else{
                    stack.push(asteroids[i]);
                }
            }
        }
        // System.out.println(stack);
        int[] arr = new int[stack.size()];
        for (int i = 0; i < arr.length; i++) {
            arr[i] = stack.get(i);
        }
        return arr;
    }
}